generator/src/main/java/com/scottlogic/deg/generator/violations/filters/ConstraintTypeViolationFilter.java

```diff
@@ -5,7 +5,7 @@
 import com.scottlogic.deg.generator.constraints.grammatical.ConditionalConstraint;
 
 public class ConstraintTypeViolationFilter implements ViolationFilter{
-    private final Class constraintType;
+    public final Class constraintType;
```

Comment:
Hmmm, why did you make this public? If you made it public just for testing purposes that was probably the wrong reason. Can you use some other way of retrieving this from the class? _Or_ if this is private do you care about it in your testing?

Reply:
changed, look at latest commit and see whether you prefer it